src/compute-client/src/service.rs

```diff
@@ -311,7 +325,32 @@ where
                             (PeekResponse::Error(e), _) => PeekResponse::Error(e),
                             (PeekResponse::Rows(mut rows), PeekResponse::Rows(r)) => {
                                 rows.extend(r.into_iter());
-                                PeekResponse::Rows(rows)
+
+                                let total_size = rows
+                                    .iter()
+                                    // Note: if the type of count changes in the future to be a
+                                    // signed integer, then we'll need to consolidate rows before
+                                    // taking this summation.
+                                    .map(|(row, count): &(Row, NonZeroUsize)| {
+                                        let size = row
+                                            .byte_len()
+                                            .saturating_add(std::mem::size_of_val(count));
+                                        u64::cast_from(size)
+                                    })
+                                    .fold(0u64, |total, x| total.saturating_add(x));
```

Comment:
Nit: If we're ever getting close to overflowing, we'd be in trouble anyways, so I think it's fine to use `sum` here, unless I'm missing something.

Reply:
That's fine with me, just wanted to make sure we didn't wrap. But you're right, if we're close to wrapping a u64 then we would have had to alreayd allocate waaaaay more memory than would be possible, will update!